### DISCUSSION TITLE
Dumping SQL during tests

### DISCUSSION THREAD

**Discussion by preslavrachev** [gobuffalo/buffalo#2273] [Q&A] [answered]
This may be a stupid question, but I spent a considerable amount of time and could not find an answer for it:

Is it possible to see the generated SQL during the execution of the tests (`buffalo test`)?

**Comment by sio4** [comment, marked as answer]
Do you mean when running `buffalo test`?

It is not directly related to the `buffalo test` command but you can configure your application to print out SQL logs when it runs ins test mode.

Please check your `app/models/models.go`. I found the following line:

```go
    pop.Debug = env == "development"
```

So basically, the pop (which handles the model/database part for the buffalo) has a variable called `Debug` and if the variable is `true`, the buffalo set `logger` (actually `pop.log`) for pop to print out its logs.

The default configuration of the newly generated app may have the above line but you can modify it as:

```go
    pop.Debug = (env == "development" || env == "test")
```

to make them print those debugging information during tests too.

**Reply by preslavrachev** [reply]
Precisely

**Reply by sio4** [reply]
I updated the original answer so it could be easier to find if someone else is searching for the same question.

By the way, I would love to hear your situation that you want to make the testing logs verbose instead of checking generated SQLs in development time. I usually suspected users want to see them while writing the code but not during the test since the logs are too verbose and probably already checked during development cycles.

Could you please let me understand what is your situation with this? That will highly help me to understand the real-life user's requirements more clearly and that would be helpful to us to improve the buffalo and pop.

**Reply by preslavrachev** [reply]
Thanks for elaborating on the answer. Indeed, this seems to solve my particular problem - maybe I will even write about it on my blog, to get some traction about Buffalo.

To give you more idea of where I come from - we use Gorm for our applications, but have been considering doing some rewrites to use Buffalo. Anyway, Gorm has this nice `.Debug()` helper method that you can slap onto a DB instance or a transaction, and it will automatically output executed SQL code to the terminal.

Is this useful during testing? Not at all times. However, it can be particularly interesting for when you are trying to run a single test in verbose mode, and want to see exactly what is going on under the hood. This was my particular case.

**Reply by sio4** [reply]
Ah! That makes sense! Understood!
